spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/data/module/deployer/cloudfoundry/AbstractPaginatedResponse.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.cloudfoundry;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+abstract class AbstractPaginatedResponse {
+
+    private volatile Integer        totalResults;
+    private volatile Integer        totalPages;
+    private volatile String         previousUrl;
+    private volatile String         nextUrl;
+    private volatile List<Resource> resources;
+
+    public final Integer getTotalResults() {
+        return totalResults;
+    }
+
+    @JsonProperty("total_results")
+    public final void setTotalResults(Integer totalResults) {
+        this.totalResults = totalResults;
+    }
+
+    public final Integer getTotalPages() {
+        return totalPages;
+    }
+
+    @JsonProperty("total_pages")
+    public final void setTotalPages(Integer totalPages) {
+        this.totalPages = totalPages;
+    }
+
+    public final String getPreviousUrl() {
+        return previousUrl;
+    }
+
+    @JsonProperty("prev_url")
+    public final void setPreviousUrl(String previousUrl) {
+        this.previousUrl = previousUrl;
+    }
+
+    public final String getNextUrl() {
+        return nextUrl;
+    }
+
+    @JsonProperty("next_url")
+    public final void setNextUrl(String nextUrl) {
+        this.nextUrl = nextUrl;
+    }
+
+    public final List<Resource> getResources() {
+        return resources;
+    }
+
+    public final void setResources(List<Resource> resources) {
+        this.resources = resources;
+    }
+
+    static final class Resource {
+
+        private volatile ListSpacesResponse.Resource.Metadata metadata;
+        private volatile ListSpacesResponse.Resource.Entity   entity;
+
+        public ListSpacesResponse.Resource.Metadata getMetadata() {
+            return metadata;
+        }
+
+        public void setMetadata(ListSpacesResponse.Resource.Metadata metadata) {
+            this.metadata = metadata;
+        }
+
+        static final class Metadata {
+
+            private volatile String id;
+            private volatile String url;
+            private volatile String createdAt;
```

Comment:
I disagree; the properties we return for a module instance (where this value surfaces) are all Strings, and I don't want any checking of this information nor conversion/reconversion from/to Strings. Besides I don't use any date-like properties of these fields. They might not even be dates, for all I know.